tests/test_execution_compute_plan.py

```diff
@@ -553,3 +553,83 @@ def test_compute_plan_no_batching(factory, client, default_dataset, default_obje
     traintuples = client.list_compute_plan_traintuples(cp.key)
     assert len(traintuples) == 2
     assert all([tuple_.status == models.Status.done for tuple_ in traintuples])
+
+
+
+LOCAL_FOLDER_ALGO_SCRIPT = f"""
+import json
+import substratools as tools
+import os
+class TestAlgo(tools.Algo):
+    def train(self, X, y, models, rank):
+        state_file = "/sandbox/local/state.json"
+        state = None
+
+        if os.path.exists(state_file):
+            with open(state_file, "r") as f:
+                state = json.load(f)
+        else:
+            state = {{ "factor": 0 }}
+
+        state["factor"] += 1
+
+        with open(state_file, "w") as f:
+            json.dump(state, f)
+
+        res = {{'value': state["factor"] * 2 }}
+
+        print(f'Train, return {{res}}')
+        return res
+
+    def predict(self, X, model):
+        res = [x * model['value'] for x in X]
+        print(f'Predict, get X: {{X}}, model: {{model}}, return {{res}}')
+        return res
+
+    def load_model(self, path):
+        with open(path) as f:
+            return json.load(f)
+
+    def save_model(self, model, path):
+        with open(path, 'w') as f:
+            return json.dump(model, f)
+
+if __name__ == '__main__':
+    tools.algo.execute(TestAlgo())
+"""
+
+@pytest.mark.slow
+@pytest.mark.remote_only
```

Comment:
This test also works in local mode, is there a reason to mark it as remote only ?

Reply:
I marked it as remote only because the first test in this file (`test_compute_plan_simple`) has the same decorator. Maybe it could be removed from there too?

Reply:
`test_compute_plan_simple` is remote only because it relies on several clients. `test_compute_plan_no_batching` should work in local mode too

Reply:
Gotcha. Fixed in https://github.com/SubstraFoundation/substra-tests/pull/186/commits/48025546fbe97e59ea72eede2891d46b73cd33b4